content/webapp/services/prismic/fetch/index.ts

```diff
@@ -80,36 +91,52 @@ export function fetcher<Document extends PrismicDocument>(
           fetchLinks,
         });
 
-        if (document.type === contentType) {
+        if (hasMatchingContentType(document, contentType)) {
           return document;
+        } else {
+          console.warn(`Asked to fetch document ${id} as type '${contentType}', but ${id} is actually '${document.type}'`);
```

Comment:
This is the first line I added when trying to debug this, and I left it in because it's a useful temperature check. If we see this a lot in the application logs, it might suggest a broken link or a bug somewhere.